include/NovelRT/Experimental/Input/IInputService.h

```diff
@@ -0,0 +1,37 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#ifndef NOVELRT_EXPERIMENTAL_INPUT_IINPUTSERVICE_H
+#define NOVELRT_EXPERIMENTAL_INPUT_IINPUTSERVICE_H
+
+#ifndef NOVELRT_EXPERIMENTAL_INPUT_H
+#error NovelRT does not support including types explicitly by default. Please include Input.h instead for the Input namespace subset.
+#endif
+
+namespace NovelRT::Experimental::Input
+{
+    class IInputService : public std::enable_shared_from_this<IInputService>
```

Comment:
I think this should be `IInputDevice` as that's more in line with our LLAPI-style conventions going forward. I also wonder if this should be an interface given the private fields, but that's a tiny nit I don't really care about changing right now.